be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/WitnessHandler.scala

```diff
@@ -1,27 +1,60 @@
 package ch.epfl.pop.pubsub.graph.handlers
 
+import akka.pattern.AskableActorRef
 import ch.epfl.pop.model.network.JsonRpcRequest
-import ch.epfl.pop.model.network.method.message.Message
 import ch.epfl.pop.model.network.method.message.data.witness.WitnessMessage
-import ch.epfl.pop.model.objects.{Hash, WitnessSignaturePair}
-import ch.epfl.pop.pubsub.graph.GraphMessage
+import ch.epfl.pop.model.objects.{Channel, DbActorNAckException, Hash, Signature}
+import ch.epfl.pop.pubsub.graph.{ErrorCodes, GraphMessage, PipelineError}
+import ch.epfl.pop.storage.DbActor
+import ch.epfl.pop.storage.DbActor.DbActorAddWitnessMessage
 
-import scala.concurrent.{Await, Future}
+import scala.concurrent.Await
+import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
 
-case object WitnessHandler extends MessageHandler {
+/**
+ * WitnessHandler object uses the db instance from the MessageHandler
+ */
+object WitnessHandler extends MessageHandler {
```

Comment:
```suggestion
object WitnessHandler {
```
I think the object does not need to inherit the trait.

Reply:
Here `MessageHandler` gives us access to the `dbActor`